rotkehlchen/chain/bitcoin_cash/__init__.py

```diff
@@ -0,0 +1,47 @@
+from typing import Dict, List
+
+import requests
+
+from rotkehlchen.chain.bitcoin_cash.utils import cash_to_legacy_address
+from rotkehlchen.errors.misc import RemoteError, UnableToDecryptRemoteData
+from rotkehlchen.fval import FVal
+from rotkehlchen.types import BTCAddress
+from rotkehlchen.utils.misc import satoshis_to_btc
+from rotkehlchen.utils.network import request_get
+
+
+def get_bitcoin_cash_addresses_balances(accounts: List[BTCAddress]) -> Dict[BTCAddress, FVal]:
+    """Queries api.haskoin.com for the balances of BCH accounts.
+
+    May raise:
+    - RemoteError if there is a problem querying api.haskoin.com
+    """
+    source = 'https://api.haskoin.com'
+    balances: Dict[BTCAddress, FVal] = {}
+    try:
+        accounts_chunks = [accounts[x:x + 80] for x in range(0, len(accounts), 80)]
+        for accounts_chunk in accounts_chunks:
+            params = ','.join(accounts_chunk)
+            bch_resp = request_get(url=f'{source}/bch/address/balances?addresses={params}')
+            for entry in bch_resp:
+                # Try and get the initial address passed to the request.
+                # This is because the API only returns CashAddr format as response.
+                if entry['address'] in accounts_chunk:
+                    balances[entry['address']] = satoshis_to_btc(FVal(entry['confirmed']))
+                else:
+                    address = cash_to_legacy_address(entry['address'])
+                    if address is not None:
+                        balances[address] = satoshis_to_btc(FVal(entry['confirmed']))
+    except (
+            requests.exceptions.RequestException,
```

Comment:
What can raise `UnableToDecryptRemoteData`?

Why catch both `requests.exceptions.RequestException` and `requests.exceptions.Timeout`? Timeout is a child class of `RequestException`. In fact `RequestException` is the mother of all exception classes in the requests library:  https://github.com/psf/requests/blob/a5e7169e759817ca309f8271c3c4fa79393a22f5/requests/exceptions.py#L61